_rules/first-focusable-elements-are-links-to-content-e53727.md

```diff
@@ -0,0 +1,61 @@
+---
+id: e53727
+name: First focusable elements are links to sections of content
+rule_type: atomic
+description: |
+  This rule checks that the first focusable elements are links referring to elements on the same page
+accessibility_requirements:
+	wcag-technique:G124: # Adding links at the top of the page to each area of the content
+		forConformance: false
+		failed: not satisfied
+		passed: satisfied
+		inapplicable: further testing needed
+input_aspects:
+	- DOM Tree
+	- CSS styling
+authors:
+	- Jean-Yves Moyen
+	- Anne Thyme Nørregard
+---
+
+## Applicability
+
+This rule applies to any [document](#https://www.w3.org/TR/dom/#concept-document) where the [document element](#https://www.w3.org/TR/dom/#document-element) is an HTML `html` element.
+
+## Expectation 1
+
+There is an [initial segment](#initial-segment) of the [focusable](#focusable) elements (in focus order) such that each element in that [initial segment](#initial-segment):
+- is [visible](#visible) when [focused](#focused); and
+- has a [semantic role](#semantic-role) of link; and
+- when activated, moves focus to a [section of content](#section-of-content) within the same [document](#https://www.w3.org/TR/dom/#concept-document); and
+- has an [accessible name](#accessible-name) that communicates that it links to that specific [section of content](#section-of-content).
+
+**Note**: There is no requirement on how many [focusable](#focusable) elements are part of that [initial segment](#initial-segment), nor any requirement to provide a way to determine (programatically or not) where that [initial segment](#initial-segment) stops. Technique [G124: Adding links at the top of the page to each area of the content](https://www.w3.org/WAI/WCAG21/Techniques/general/G124), and this rule, only require that such an set exists.
+
+**Editorial note**: An attempt to clarify what this initial segment has to be is done in the second expectation: it must contain exactly one link for each section of content in the page. I am still not very happy with the formulation, nor with the order of these two expectations. Any suggestions to improve that are welcome…
+
+## Expectation 2
+
+Each [section of content](#section-of-content) in the [document](#https://www.w3.org/TR/dom/#concept-document) is the target of exactly one link from the set of [focusable](#focusable) elements that passes Expectation 1.

```

Comment:
With this expectation, you're stating that all sections of content in the document need to be a target of one the links. I'm not sure that that is implied in G124, and I don't think that needs to happen (you probably don't want to fail a page that does not have skip links to the adverts section). This can be tackled in the definition of section of content (by explicitly leaving out specific sections of content, like advertisement), or this expectation can be rewritten to state that each link in the initial segment targets a different section of content.

Reply:
Yes, the difficulty is to make G124 different from G1… Since G124 doesn't specify the number of links, it looks like anything that passes G1 also passes G124 (with a set of 1 link). I think the main difference is that G124 states:
>  by providing a list of links to *the different* section*s* of the content.

(emphasise mine)
which I interpreted as "to each of them", as opposed to G1 which is only "to the main content".

The, I agree that I don't want to fail a page that "misses" a link to Ads… And trying to put that into the definition of "Section of content" might spiral into an endless list of exceptions.

Saying that the links go to different sections is not very good for separating G124 from G1: a single link to main would pass that also.

I understand that a page with a navigation bar and a main content and only a link to main content should pass G1 but fail G124. The challenge here is to manage to write the rules in a way that it is possible to pass (the rule for) G1 but fail (the rule for) G124.

Requiring several links for G124 is not really good either since I also understand that if a page only has main content, then a single link to it would pass both G1 and G124 (and probably be a bit pointless since there are no blocks to bypass, but that's another issue).

Reply:
I agree with your argument. Per the description of G124 "each area of the content" will need to be linked. But this means that we are going to have to be extra careful when writing the definition of "section of content", which is left undefined in WCAG, unfortunately

Reply:
Yes. We have a definition of [Section of content](https://act-rules.github.io/glossary/#section-of-content) which does not rule ads or similar stuff (it even includes it since ads serve a purpose…)

Reply:
I had forgotten we already had that definition :-)
Let's see what others think about linking to each section of content.